SystemInformer/mainwnd.c

```diff
@@ -1363,7 +1347,7 @@ VOID PhMwpOnCommand(
                 L"Current Window Desktop",
                 L"Desktop",
                 PhpOpenSecurityDesktopHandle,
-                PhpCloseSecurityDesktopHandle,
+                NULL,
```

Comment:
🤔 is this not going to leak the handle (and below)? Also, how did this ever work? `PhSecurityInformation_Release` passes the `Context` to the `CloseObject` and not the `Handle`? Maybe it was intended for `PhpOpenSecurityStationHandle` to set the output context pointer to the opened handle?

Reply:
`Handle` is opened from `PhStdGetObjectSecurity` via `OpenObject`. After retrieving of security info by `PhGetObjectSecurity` `Handle` is closing by `NtClose` inside `PhStdGetObjectSecurity`. CloseObject only useful for free resources after security window closes (only Object Manager uses it). Previous `PhpCloseSecurityDesktopHandle` and `PhpCloseSecurityStationHandle` code hadn't effect since `Context` was `NULL`.

If Desktop handle was open by `OpenDesktop`, `NtClose` can't close it and returns `STATUS_HANDLE_NOT_CLOSABLE` (despite handle doesn't have the Protected attribute). But this doesn't happen if handle was duplicated (seems like it's a bug in Windows?). To fix this I added this code to `PhStdGetObjectSecurity`.

```
        if (NtClose(handle) == STATUS_HANDLE_NOT_CLOSABLE &&
            PhEqualString2(this->ObjectType, L"Desktop", TRUE))
        {
            CloseDesktop(handle);
        } 
```

Example of huge Desktop handles leak in current release build:

https://github.com/user-attachments/assets/045f0ae0-2511-4ece-897e-1103ac035a7e

Reply:
> NtClose can't close it and returns STATUS_HANDLE_NOT_CLOSABLE (despite handle doesn't have the Protected attribute).

That's because it's not in the handle table of the process it's in the win32k subsystem. It's errant to try to close a desktop handle with `NtClose`. They need to go through win32k `NtUserCloseDesktop` (`CloseDesktop`) must be used. Also passing a value to `NtClose` to "try" to close the handle has the possibility of closing the wrong handle. They shouldn't be mixed.

Reply:
Added a comment elsewhere for a case that wasn't covered. Thanks for the explanation 👍 .